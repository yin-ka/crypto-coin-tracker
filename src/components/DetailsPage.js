import { useLocation, useNavigate } from 'react-router-dom';
import { BiArrowBack } from 'react-icons/bi';
import { FaChevronDown, FaChevronUp, FaBitcoin } from 'react-icons/fa';
import styles from '../styles/Details.css';

const DetailsPage = () => {
  const location = useLocation();
  const navigate = useNavigate();

  const backHome = () => {
    navigate('/');
  };

  const { coins } = location.state || { coins: {} };

  return (
    <div>
      <div>
        <BiArrowBack onClick={() => backHome()} className={styles.icon} />
      </div>
      <div className={styles.container}>
        <h1 className={styles.name}>
          {coins.name}
          {' '}
          (
          {coins.symbol}
          )
        </h1>
        <div className={styles.coin}>
          <p className={styles.cap}>
            market cap: $
            {Number(coins.market_cap_usd).toLocaleString(undefined, {
              maximumFractionDigits: 2,
            })}
          </p>
          <p className={styles.volume}>
            volume(24h): $
            {Number(coins.volume24a).toLocaleString(undefined, {
              maximumFractionDigits: 2,
            })}
          </p>
          <p className={styles.btc}>
            price(btc):
            <span className={styles.span}>
              {' '}
              <FaBitcoin />
              {' '}
              {coins.price_btc}
            </span>
          </p>
          <p className={styles.usd}>
            price(usd): $
            {Number(coins.price_usd).toLocaleString(undefined, {
              maximumFractionDigits: 2,
            })}
          </p>
          <p className={styles.hour}>
            1h %:
            {' '}
            {coins.percent_change_1h < 0 ? (
              <span className={styles.span}>
                <FaChevronDown color="red" />
                {Math.abs(coins.price_btc)}
              </span>
            ) : (
              <span className={styles.span}>
                <FaChevronUp color="green" />
                {Math.abs(coins.price_btc)}
              </span>
            )}
          </p>
          <p className={styles.day}>
            24h %:
            {' '}
            {coins.percent_change_24h < 0 ? (
              <span className={styles.span}>
                <FaChevronDown color="red" />
                {Math.abs(coins.price_btc)}
              </span>
            ) : (
              <span className={styles.span}>
                <FaChevronUp color="green" />
                {Math.abs(coins.price_btc)}
              </span>
            )}
          </p>
          <p className={styles.week}>
            7d %:
            {' '}
            {coins.percent_change_24h < 0 ? (
              <span className={styles.span}>
                <FaChevronDown color="red" />
                {Math.abs(coins.price_btc)}
              </span>
            ) : (
              <span className={styles.span}>
                <FaChevronUp color="green" />
                {Math.abs(coins.price_btc)}
              </span>
            )}
          </p>
          <p className={styles.supply}>
            volume(24h):
            {' '}
            {Number(coins.tsupply).toLocaleString(undefined, {
              maximumFractionDigits: 2,
            })}
          </p>
        </div>
      </div>
    </div>
  );
};
export default DetailsPage;
