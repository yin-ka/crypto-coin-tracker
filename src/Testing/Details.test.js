import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import DetailsPage from '../components/DetailsPage';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Details page', () => {
  test('Render Details Page Component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <DetailsPage />
          </Provider>
          ,
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
