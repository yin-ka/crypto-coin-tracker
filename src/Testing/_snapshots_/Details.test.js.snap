// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page Render Details Page Component 1`] = `
Array [
  <div>
    <div>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 11H6.414l5.293-5.293-1.414-1.414L2.586 12l7.707 7.707 1.414-1.414L6.414 13H21z"
        />
      </svg>
    </div>
    <div
      className="container"
    >
      <h1
        className="name"
      >
         
        (
        )
      </h1>
      <div
        className="coin"
      >
        <p
          className="cap"
        >
          market cap: $
          NaN
        </p>
        <p
          className="volume"
        >
          volume(24h): $
          NaN
        </p>
        <p
          className="btc"
        >
          price(btc):
          <span
            className="span"
          >
             
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zm-141.651-35.33c4.937-32.999-20.191-50.739-54.55-62.573l11.146-44.702-27.213-6.781-10.851 43.524c-7.154-1.783-14.502-3.464-21.803-5.13l10.929-43.81-27.198-6.781-11.153 44.686c-5.922-1.349-11.735-2.682-17.377-4.084l.031-.14-37.53-9.37-7.239 29.062s20.191 4.627 19.765 4.913c11.022 2.751 13.014 10.044 12.68 15.825l-12.696 50.925c.76.194 1.744.473 2.829.907-.907-.225-1.876-.473-2.876-.713l-17.796 71.338c-1.349 3.348-4.767 8.37-12.471 6.464.271.395-19.78-4.937-19.78-4.937l-13.51 31.147 35.414 8.827c6.588 1.651 13.045 3.379 19.4 5.006l-11.262 45.213 27.182 6.781 11.153-44.733a1038.209 1038.209 0 0 0 21.687 5.627l-11.115 44.523 27.213 6.781 11.262-45.128c46.404 8.781 81.299 5.239 95.986-36.727 11.836-33.79-.589-53.281-25.004-65.991 17.78-4.098 31.174-15.792 34.747-39.949zm-62.177 87.179c-8.41 33.79-65.308 15.523-83.755 10.943l14.944-59.899c18.446 4.603 77.6 13.717 68.811 48.956zm8.417-87.667c-7.673 30.736-55.031 15.12-70.393 11.292l13.548-54.327c15.363 3.828 64.836 10.973 56.845 43.035z"
              />
            </svg>
             
          </span>
        </p>
        <p
          className="usd"
        >
          price(usd): $
          NaN
        </p>
        <p
          className="hour"
        >
          1h %:
           
          <span
            className="span"
          >
            <svg
              color="green"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "green",
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              />
            </svg>
            NaN
          </span>
        </p>
        <p
          className="day"
        >
          24h %:
           
          <span
            className="span"
          >
            <svg
              color="green"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "green",
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              />
            </svg>
            NaN
          </span>
        </p>
        <p
          className="week"
        >
          7d %:
           
          <span
            className="span"
          >
            <svg
              color="green"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "green",
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              />
            </svg>
            NaN
          </span>
        </p>
        <p
          className="supply"
        >
          volume(24h):
           
          NaN
        </p>
      </div>
    </div>
  </div>,
  ",",
]
`;